src/Simplex/Chat/Store/Groups.hs

```diff
@@ -1719,6 +1800,58 @@ getForwardScopeMember db vr user GroupMember {groupMemberId = sendingGMId} scope
         (sendingGMId, scopeGMId, scopeGMId, sendingGMId, GMIntroReConnected, GMIntroToConnected, GMIntroConnected)
   pure introExists_ $>> (eitherToMaybe <$> runExceptT (getGroupMemberById db vr user scopeGMId))
 
+getIntroducedRelations :: DB.Connection -> GroupMemberId -> IO [(Int64, MemberRelation)]
+getIntroducedRelations db gmId =
+  map ((,MRIntroduced) . fromOnly) <$>
+    DB.query
+      db
+      [sql|
+        SELECT m.index_in_group
+        FROM group_member_intros i
+        JOIN group_members m ON m.group_member_id = i.re_group_member_id
+        WHERE i.to_group_member_id = ? AND i.intro_status NOT IN (?,?,?)
+      |]
+      (gmId, GMIntroReConnected, GMIntroToConnected, GMIntroConnected)
+
+getIntroducedToRelations :: DB.Connection -> GroupMemberId -> IO [(Int64, MemberRelation)]
+getIntroducedToRelations db gmId =
+  map ((,MRIntroducedTo) . fromOnly) <$>
+    DB.query
+      db
+      [sql|
+        SELECT m.index_in_group
+        FROM group_member_intros i
+        JOIN group_members m ON m.group_member_id = i.to_group_member_id
+        WHERE i.re_group_member_id = ? AND i.intro_status NOT IN (?,?,?)
+      |]
+      (gmId, GMIntroReConnected, GMIntroToConnected, GMIntroConnected)
+
+getIntroConnectedRelations :: DB.Connection -> GroupMemberId -> IO [(Int64, MemberRelation)]
+getIntroConnectedRelations db gmId = do
+  introducedIdxs <-
+    map ((,MRConnected) . fromOnly) <$>
+      DB.query
+        db
+        [sql|
+          SELECT m.index_in_group
+          FROM group_member_intros i
+          JOIN group_members m ON m.group_member_id = i.re_group_member_id
+          WHERE i.to_group_member_id = ? AND i.intro_status IN (?,?,?)
+        |]
+        (gmId, GMIntroReConnected, GMIntroToConnected, GMIntroConnected)
+  invitedIdxs <-
+    map ((,MRConnected) . fromOnly) <$>
+      DB.query
+        db
+        [sql|
+          SELECT m.index_in_group
+          FROM group_member_intros i
+          JOIN group_members m ON m.group_member_id = i.to_group_member_id
+          WHERE i.re_group_member_id = ? AND i.intro_status IN (?,?,?)
+        |]
+        (gmId, GMIntroReConnected, GMIntroToConnected, GMIntroConnected)
```

Comment:
why do they need to be two queries and not one query with OR in join?

JOIN group_members m ON m.group_member_id = i.re_group_member_id OR m.group_member_id = i.to_group_member_id